src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -21,9 +53,12 @@ - (instancetype) initWithParams:(NSString *)name
     _styles = styles;
     
     _currentStyle = 0;  
-  
-    _gamma_weights = gammaWeights;
-    _beta_weights = betaWeights;
+    
+    _gamma_weights = (float *) malloc(numberFeatureChannels * styles * sizeof(float));
```

Comment:
Hmm, seeing more of this in context, I think it's probably clearest and safest just to use `NSMutableData` to allocate and own this memory. So you could make `_gamma_weights` an `NSMutableData *` and initialize as e.g.:
```
_gamma_weights = [NSMutableData dataWithLength:numberFeatureChannels * styles * sizeof(float)];
```
Then you can get to the memory using `_gamma_weights.mutableBytes`.

See https://developer.apple.com/documentation/foundation/nsmutabledata?language=objc 

With NSData, you don't have to worry about implementing `dealloc`, and then you can just pass `mutableBytes` into the MPS API

Reply:
Perfect, updated the implementation to use NSMutableData